nemo/collections/asr/parts/utils/transcribe_utils.py

```diff
@@ -299,6 +310,21 @@ def prepare_audio_data(cfg: DictConfig) -> Tuple[List[str], bool]:
     return filepaths, partial_audio
 
 
+def restore_transcription_order(manifest_path: str, transcriptions: list) -> list:
```

Comment:
minor: If possible we can do this while writing the manifest to avoid creating another list in memory (this will help when dealing with large dataset inferences).

Reply:
Agreed but we don't control who writes manifest so we still need this for those which are externally written

Reply:
not sure that it can be entirely avoided but I added one `del` that should ease the CPU memory requirements for these cases.